lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceManagerStatus.java

```diff
@@ -33,43 +33,54 @@
 import java.util.logging.Logger;
 
 /**
- * Manages the status of the Resource Manager.
+ * Manages the status of the Resource Manager and tracks whether it is idle.
  */
 @DriverSide
 @Private
-public final class ResourceManagerStatus implements EventHandler<RuntimeStatusEvent>,
-    DriverIdlenessSource {
+public final class ResourceManagerStatus implements EventHandler<RuntimeStatusEvent>, DriverIdlenessSource {
+
   private static final Logger LOG = Logger.getLogger(ResourceManagerStatus.class.getName());
 
   private static final String COMPONENT_NAME = "ResourceManager";
+
   private static final IdleMessage IDLE_MESSAGE =
       new IdleMessage(COMPONENT_NAME, "No outstanding requests or allocations", true);
 
   private final ResourceManagerErrorHandler resourceManagerErrorHandler;
   private final DriverStatusManager driverStatusManager;
   private final InjectionFuture<DriverIdleManager> driverIdleManager;
 
-  // Mutable state.
-  private State state =  State.INIT;
+  /** Mutable RM state. */
+  private State state = State.INIT;
+
+  /** Number of container requests pending with the RM, as per latest RuntimeStatusEvent message. */
   private int outstandingContainerRequests = 0;
```

Comment:
I'd rather update the comment to say "number of outstanding container requests..." to minimize the change 😄